src/lib/settings/group/SettingsArray.js

```diff
@@ -0,0 +1,127 @@
+const { resolveGuild, arraysStrictEquals, mergeObjects, makeObject } = require('../../util/util');
+
+const checkForIndex = (value) => Array.isArray(value) && value.length === 2 && typeof value[0] === 'number';
+
+class SettingsArray {
+	constructor(entry) {
+		Object.defineProperty(this, 'base', { value: null, writable: true });
+		
+		// The Entry this SettingArray refers to
+		Object.defineProperty(this, 'entry', { value: entry });
+
+		this.data = entry.default;
+	}
+
+	get gateway() {
+		return this.base.gateway;
+	}
+
+	get(index) {
+		if (index !== undefined && index !== null) {
+			return this.data[index];
+		}
+		return this.data.slice();
+	}
+
+	includes(value) {
+		return this.data.includes(value);
+	}
+
+	async update(values, options) {
+		if (!Array.isArray(values)) values = [values];
+
+		const guild = resolveGuild(this.base.gateway.client, 'guild' in options ? options.guild : this.base.target);
+		const { entry } = this;
+		let indexing = false;
+
+		// Not sure of a better way to do this, come back at a later time
+		if (values.some(checkForIndex)) {
+			if (!(values.every(checkForIndex))) throw "Indexing found. You must only use straight indexing or no indexing, not a mixture of both.";
+			indexing = true;
+		}
+
+		const { errors, clone } = await this._parse(values, options, guild, indexing);
+		// This might need to be changed/adjusted to give the user better throw behavior.
+		if (errors.length) throw { errors, updated: [] };
+		if (arraysStrictEquals(this.data, clone)) return { errors: [], updated: [] };
+		
+		const result = [{ key: entry.path, value: clone, entry }];
+		this._save(result);
+		
+		return { errors, updated: result };
+	}
+
+	async _parse(values, options, guild, indexing) {
+		const { serializer: { serialize } } = this.entry;
+
+		if (indexing) {
+			values = await Promise.all(values.map(async ([i, v]) => ([i, serialize(await entry.parse(v, guild))])));
+		} else if (!Array.isArray(values)) {
+			values = serialize(await entry.parse(values, guild));
+		} else values = await Promise.all(values.map(async val => serialize(await entry.parse(val, guild))));
+
+		const { action = 'auto' } = options;
+		if (!indexing && action === 'overwrite') return values;
+
+		const clone = this.get();
+
+		// Errors are given back in the order that values were sent in
+		const errors = [];
+
+		// This value has an index paired with it
+		if (indexing) {
+			for (const val of values) {
+				let index = val[0];
```

Comment:
Could use some destructuring, use it at the bottom

```suggestion
				let [index, actualValue] = val;
```